### PR TITLE
api: add support for deleting apps

### LINKED ISSUE #274: Support app deletion API
It appears that users can create an app, but they are not capable to remove it.
This ticket is about implementing an API call HTTP delete against /apps/{app}

Flow:
1. Check if app has routes.
2. If not - delete an app
3. If yes - forbid.

### PR DESCRIPTION
**Pull request by ccirello**
Fixes #274

### CODE REVIEW

**Comment by ccirello** [comment]
Depends on #325

File: api/models/app.go [hunk 1]
```diff
@@ -8,15 +8,16 @@ import (
...
-	ErrAppsRemoving        = errors.New("Could not remove app from datastore")
 	ErrAppsGet             = errors.New("Could not get app from datastore")
 	ErrAppsList            = errors.New("Could not list apps from datastore")
-	ErrAppsAlreadyExists   = errors.New("App already exists")
+	ErrAppsMissingNew      = errors.New("Missing new application")
 	ErrAppsNotFound        = errors.New("App not found")
 	ErrAppsNothingToUpdate = errors.New("Nothing to update")
-	ErrAppsMissingNew      = errors.New("Missing new application")
+	ErrAppsRemoving        = errors.New("Could not remove app from datastore")
+	ErrAppsPendingRoutes   = errors.New("App with pending routes")
```
Comment (treeder) [inline api/models/app.go]: What do you mean by pending routes?

Comment (treeder) [inline api/models/app.go]: Does this just mean it has routes?

Comment (ccirello) [inline api/models/app.go]: > Does this just mean it has routes?

Gonna improve the name.

Comment (ccirello) [inline api/models/app.go]: Done

**Pull request merged**